### DISCUSSION TITLE
PowerShell 5 & PowerShell 7 Module version incompatabilities

### DISCUSSION THREAD

**Discussion by msnyder67** [PowerShell/PowerShell#20917] [Q&A]
Am I missing something or is it time for PS 7.x to have its own Module Path and environment variable?

I am doing a lot of coding in PowerShell.
Some of the Scripts require features available only in Power Shell 7. (Invoke-RestMethod parameters etc)
The problem I am running into is Module PS version incompatibility.

As of late, a lot of the updated modules are only being released for PowerShell 7.x
So I install a Module to get something to work in PS 7.x and Break the functionality in PS 5.x.
Fixing the functionality in PS 5.x (uninstalling the New module and re-installing the older compatible version) breaks PS 7.x
I know that PS supports installing multiple versions of a module, but my experience suggests that PS X.x does know which version to use

I am sure someone will purpose that I should just use PS 7.x.
Hard to do when the Get-InstalledModule and Install-Module commands appear to do nothing in a PS 7.3+ (I am seeing this on multiple machines, Workstations and Servers)

**Comment by rhubarb-geek-nz** [comment]
With PowerShell 5.1

```
D:\OneDrive\Documents\WindowsPowerShell\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
```

With PowerShell 7.3.10

```
D:\OneDrive\Documents\PowerShell\Modules;C:\Program Files\PowerShell\Modules;c:\program files\powershell\7\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
```

The PowerShell 5.1 directories have WindowsPowerShell as part of their directory naming.

So what should happen is that PowerShell 7.x first looks in it's directories first before falling back and trying to use a desktop module

Likewise, if you install a module with PowerShell 7.x it should put it in the first directory in the path, which should belong to it

Also, the manifests for the modules should say if they support 'Core' or 'Desktop' and what version. So it would be nice if PowerShell ignored modules that were not for it.
